### DISCUSSION TITLE
Making IManagementClient.GetAsync (and Put, Post, Delete) APIs public.

### DISCUSSION THREAD

**Discussion by inikulshin** [EasyNetQ/EasyNetQ.Management.Client#284] [Ideas]
It'll help developers to use ManagementClient even in cases, EasyNetQ doesn't implement some RabbitMQ Management HTTP API (e.g. new   `/api/stream/...` of `rabbitmq_stream_management` plugin), or implements some API partially (e.g. PolicyDefinition before #281), or implements API for specific RabbitMQ versions only.

It's essential for developer's custom management plugins that may extend basic RabbitMQ Management HTTP API (like `rabbitmq_stream_management`, `rabbitmq_shovel_management` or `rabbitmq_federation_management`).

And last, but not the least: it may reduce `IManagementClient` public API to O(1) methods, making all current public methods just an extensions.

**Comment by inikulshin** [comment]
@Pliner @zidad @micdenny @luigiberrettini FYI

**Reply by zidad** [reply]
Would be fine my me. I would also be interested in changing the management client to be more in line with a statically typed HTTP client, so the constructor can be simplified: 

https://timdeschryver.dev/blog/refactor-your-net-http-clients-to-typed-http-clients

**Comment by inikulshin** [comment]
Implemented in https://github.com/EasyNetQ/EasyNetQ.Management.Client/pull/319
